src/Text/Parsing/Parser.purs

```diff
@@ -123,8 +125,14 @@ consume :: forall s m. Monad m => ParserT s m Unit
 consume = modify \(ParseState input position _) ->
   ParseState input position true
 
+-- | Returns the current position in the stream.
+position :: forall s m. Monad m => ParserT s m Position
+position = gets \(ParseState _ pos _) -> pos
+
 -- | Fail with a message.
 fail :: forall m s a. Monad m => String -> ParserT s m a
-fail message = do
-  position <- gets \(ParseState _ pos _) -> pos
-  throwError (ParseError message position)
+fail message = throwError <<< ParseError message =<< position
+
+-- | Fail with a message and a position.
+failWithPosition :: forall m s a. Monad m => String -> Position -> ParserT s m a
+failWithPosition message position = throwError (ParseError message position)
```

Comment:
Did you intend to use this in `fail` or something? Currently seems unused within the library at least.